Mathlib/RingTheory/Valuation/RankOne.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Data.Real.NNReal
+import Mathlib.RingTheory.Valuation.Basic
+
+/-!
+# Rank one valuations
+
+We define rank one valuations.
+
+## Main Definitions
+* `IsRankOne` : A valuation has rank one if it is nontrivial and its image is contained in `ℝ≥0`.
+
+## Tags
+
+valuation, rank one
+-/
+
+noncomputable section
+
+open Function Multiplicative
+
+open scoped NNReal
+
+variable {R : Type*} [Ring R] {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+
+namespace Valuation
+
+/-- A valuation has rank one if it is nontrivial and its image is contained in `ℝ≥0`. -/
```

Comment:
There should be a note in the docstring that this class contains the data of an injection of the target monoid into NNReal, so it's not a Prop.

If we want `IsRankOne`, it's just `Nonempty (RankOne)`.